fedimint-core/src/amount.rs

```diff
@@ -105,96 +93,58 @@ impl Amount {
         self.msats as f64 / 1000.0
     }
 
-    pub fn checked_sub(self, other: Self) -> Option<Self> {
-        Some(Self {
-            msats: self.msats.checked_sub(other.msats)?,
-        })
-    }
-
     pub fn checked_add(self, other: Self) -> Option<Self> {
         Some(Self {
             msats: self.msats.checked_add(other.msats)?,
         })
     }
-}
-
-impl std::fmt::Display for Amount {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{} msat", self.msats)
-    }
-}
-
-impl std::ops::Rem for Amount {
-    type Output = Self;
 
-    fn rem(self, rhs: Self) -> Self::Output {
+    pub fn saturating_add(self, other: Self) -> Self {
         Self {
-            msats: self.msats % rhs.msats,
+            msats: self.msats.saturating_add(other.msats),
         }
     }
-}
-
-impl std::ops::RemAssign for Amount {
-    fn rem_assign(&mut self, rhs: Self) {
-        self.msats %= rhs.msats;
-    }
-}
-
-impl std::ops::Div for Amount {
-    type Output = u64;
 
-    fn div(self, rhs: Self) -> Self::Output {
-        self.msats / rhs.msats
+    pub fn saturating_add_to(&mut self, other: Self) {
+        self.msats = self.msats.saturating_add(other.msats);
     }
-}
 
-impl std::ops::SubAssign for Amount {
-    fn sub_assign(&mut self, rhs: Self) {
-        self.msats -= rhs.msats;
+    pub fn checked_sub(self, other: Self) -> Option<Self> {
+        Some(Self {
+            msats: self.msats.checked_sub(other.msats)?,
+        })
     }
-}
 
-impl std::ops::Mul<u64> for Amount {
-    type Output = Self;
-
-    fn mul(self, rhs: u64) -> Self::Output {
+    pub fn saturating_sub(self, other: Self) -> Self {
         Self {
-            msats: self.msats * rhs,
+            msats: self.msats.saturating_sub(other.msats),
         }
     }
-}
 
-impl std::ops::Mul<Amount> for u64 {
-    type Output = Amount;
-
-    fn mul(self, rhs: Amount) -> Self::Output {
-        Amount {
-            msats: self * rhs.msats,
-        }
+    pub fn checked_mul(self, other: u64) -> Option<Self> {
+        Some(Self {
+            msats: self.msats.checked_mul(other)?,
+        })
     }
-}
 
-impl std::ops::Add for Amount {
-    type Output = Self;
-
-    fn add(self, rhs: Self) -> Self::Output {
+    pub fn saturating_mul(self, other: u64) -> Self {
         Self {
-            msats: self.msats + rhs.msats,
+            msats: self.msats.saturating_mul(other),
         }
     }
-}
 
-impl std::ops::AddAssign for Amount {
-    fn add_assign(&mut self, rhs: Self) {
-        *self = *self + rhs;
+    pub fn checked_div(self, other: Self) -> Option<u64> {
```

Comment:
Note for reviewer: We're not adding `saturating_div()` because, for unsigned integers, it would be impossible to overflow